lib/commons/dom/is-skip-link.js

```diff
@@ -0,0 +1,36 @@
+/* global dom */
+
+// test for hrefs that start with # or /# (for angular)
+const isInternalLinkRegex = /^\/?#[^/!]/;
+
+/**
+ * Determines if element is a skip link
+ * @method isSkipLink
+ * @memberof axe.commons.dom
+ * @instance
+ * @param  {Element} element
+ * @return {Boolean}
+ */
+dom.isSkipLink = function(element) {
+	if (!isInternalLinkRegex.test(element.getAttribute('href'))) {
+		return false;
+	}
+
+	// define a skip link as any anchor element whose href starts with `#...`
+	// and which precedes the first anchor element whose href doesn't start
+	// with  `#...` (that is, a link to a page)
+	const firstPageLink = axe.utils.querySelectorAll(
+		axe._tree,
+		'a:not([href^="#"]):not([href^="/#"]):not([href^="javascript"])'
+	)[0];
+	if (firstPageLink) {
```

Comment:
Nitpick: In the spirit of keeping the [happy path on the left](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88), I'd suggest `if (!firstPageLink) { return true }`. Not having a first link is an exception, not the default.